webui/react/src/components/RunFilterInterstitialModalComponent.tsx

```diff
@@ -0,0 +1,149 @@
+ *
+ */
+export const RunFilterInterstitialModalComponent = forwardRef<ControlledModalRef, Props>(
+  ({ projectId, selection, filterFormSet, onCloseAction }: Props, ref): JSX.Element => {
+    const InterstitialModal = useModal(InterstitialModalComponent);
+    const [isOpen, setIsOpen] = useState<boolean>(false);
+    const closeController = useRef(new AbortController());
+
+    const { close: internalClose, open: internalOpen } = InterstitialModal;
+
+    const open = () => {
+      setIsOpen(true);
+      internalOpen();
+    };
+
+    const close = useCallback(() => {
+      setIsOpen(false);
+      // encourage render with isOpen to false before closing to prevent
+      // firing onCloseAction twice
+      setTimeout(() => internalClose('close'), 0);
+      closeController.current.abort();
+      closeController.current = new AbortController();
+    }, [internalClose]);
+
+    useImperativeHandle(ref, () => ({ close, open }));
+
+    const selectionHasSearchRuns = useAsync(
+      async (canceler) => {
+        if (!isOpen) return NotLoaded;
+        const mergedCanceler = mergeAbortControllers(canceler, closeController.current);
+        const idToFilter = (operator: Operator, id: number) =>
+          ({
+            columnName: 'id',
+            kind: 'field',
+            location: 'LOCATION_TYPE_RUN',
+            operator,
+            type: 'COLUMN_TYPE_NUMBER',
+            value: id,
+          }) as const;
+        const filterGroup: FilterFormSetWithoutId['filterGroup'] =
+          selection.type === 'ALL_EXCEPT'
+            ? {
+                children: [
+                  filterFormSet.filterGroup,
+                  {
+                    children: selection.exclusions.map(idToFilter.bind(this, '!=')),
+                    conjunction: 'or',
+                    kind: 'group',
+                  },
+                ],
+                conjunction: 'and',
+                kind: 'group',
+              }
+            : {
+                children: selection.selections.map(idToFilter.bind(this, '=')),
+                conjunction: 'or',
+                kind: 'group',
+              };
+        const filter: FilterFormSetWithoutId = {
+          ...filterFormSet,
+          filterGroup: {
+            children: [
+              filterGroup,
+              {
+                columnName: 'numTrials',
+                kind: 'field',
+                location: 'LOCATION_TYPE_EXPERIMENT',
+                operator: '>',
+                type: 'COLUMN_TYPE_NUMBER',
+                value: 1,
+              } as const,
+            ],
+            conjunction: 'and',
+            kind: 'group',
+          },
+        };
+        try {
+          const results = await searchRuns(
+            {
+              filter: JSON.stringify(filter),
+              limit: 0,
+              projectId,
+            },
+            { signal: mergedCanceler.signal },
+          );
+
+          return (results.pagination.total || 0) > 0;
```

Comment:
this should be wrapped by `Loaded`?

Reply:
useAsync wraps the value for us. we only need to return a loadable when we want to force the value to return `NotLoaded` or `Failed`.